src/services/ApiService.ts

```diff
@@ -0,0 +1,41 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { IBook, IResponse } from "../interfaces";
+
+const BASE_URL = "https://stapi.co/api/v2/rest";
+
+export const api = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({ baseUrl: BASE_URL }),
+  tagTypes: ["Books"],
+  endpoints: (builder) => ({
+    getBooks: builder.query<IResponse, { searchTerm: string; page?: number }>({
+      query: ({ searchTerm, page = 0 }) => ({
+        url: `/book/search?pageNumber=${page}&pageSize=10`,
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/x-www-form-urlencoded",
+        },
+        body: `title=${searchTerm}`,
+      }),
+      transformResponse: (response: IResponse) => ({
+        books: response.books || [],
+        totalPages: response.page.totalPages || 0,
+        page: response.page,
+        sort: response.sort,
+      }),
```

Comment:
```suggestion
      transformResponse: ({books, page, sort}: IResponse) => ({
        books,
        totalPages: page.totalPages,
        page,
        sort,
      }),
```
апишка если ничего не находит, то все равно вернет тебе [] и 0. Если же ты хочешь защититься от некореектного ответа апишки, то на строчке `totalPages: response.page.totalPages || 0,` ты обрабатываешь кейс, в котором totalPages === undefined. если response или page будет undefined то ты словишь ошибку. 

ну или response?.page?.totalPages || 0 но предлагаю просто забить. вроде апиха корректно отрабатывает

Reply:
кстати transformResponse можно в отдельный файл рядом выносить. но т.к. он один у тебя, то ок